code/headers/gas_sensor_interface_c.hpp

```diff
@@ -4,38 +4,28 @@
 
 namespace r2d2::gas_detection {
 
-    /**
-     * Class gas_sensor_interface_base_c is needed to create an array of
-     * gas_sensor_interface_c instances. We cant do this directly because
-     * gas_sensor_interface_c is an templated class.
-     */
-    class gas_sensor_interface_base_c {
-        /**
-         * This function calibrates the sensor.
-         */
-        virtual void calibrate() = 0;
-    };
-
     /**
      * Class gas_sensor_interface_c provides an interface of all the gas
      * sensors.
-     *
      */
     template <int AmountOfGasses>
-    class gas_sensor_interface_c : public gas_sensor_interface_base_c {
+    class gas_sensor_interface_c {
     public:
-        gas_sensor_interface_c() {
-        }
 
         /**
-         *  This funtion gets the return value from the gas sensor.
+         *  This function gets the return value from the gas sensor.
```

Comment:
This is a method, because it is inside a class. In my opinion function should be replaced with method

Reply:
@YouriMulder It is technically a method, but isn't the word function more used usually? i doubt if i have to change all comments saying function to method

Reply:
@YouriMulder I dont know if its on purpose but in the style guide, the methods documentation are also using the word "function"